src/lib/schematics/update/rules/switchIdentifiersRule.ts

```diff
@@ -8,135 +8,103 @@
+    // developer's application.
+    if (isNamespaceImportNode(identifier) && isMaterialImportDeclaration(identifier)) {
+      this.trustedNamespaces.add(identifier.text);
 
-    // If the symbol is not defined or could not be resolved, just skip the following identifier
-    // checks.
-    if (!symbol || !symbol.name || symbol.name === 'unknown') {
-      console.error(`Could not resolve symbol for identifier "${identifier.text}" ` +
-        `in file ${this._getRelativeFileName()}`);
-      return;
+      return this._createFailureWithReplacement(identifier);
     }
 
-    // For export declarations that are referring to Angular Material, the identifier should be
-    // switched to the new name.
+    // For export declarations that are referring to Angular Material or the CDK, the identifier
+    // can be immediately updated to the new name.
     if (isExportSpecifierNode(identifier) && isMaterialExportDeclaration(identifier)) {
-      return this.createIdentifierFailure(identifier, symbol);
+      return this._createFailureWithReplacement(identifier);
     }
 
-    // For import declarations that are referring to Angular Material, the value declarations
-    // should be stored so that other identifiers in the file can be compared.
+    // For import declarations that are referring to Angular Material or the CDK, the name of
+    // the import identifiers. This allows us to identify identifiers that belong to Material and
+    // the CDK, and we won't accidentally touch a developer's identifier.
     if (isImportSpecifierNode(identifier) && isMaterialImportDeclaration(identifier)) {
-      this.materialDeclarations.push(symbol.valueDeclaration);
+      this.trustedIdentifiers.add(identifier.text);
 
-    // For identifiers that are not part of an import or export, the list of Material declarations
-    // should be checked to ensure that only identifiers of Angular Material are updated.
-    // Identifiers that are imported through an Angular Material namespace will be updated.
-    } else if (this.materialDeclarations.indexOf(symbol.valueDeclaration) === -1 &&
-              !this._isIdentifierFromNamespace(identifier)) {
-      return;
+      return this._createFailureWithReplacement(identifier);
     }
 
-    return this.createIdentifierFailure(identifier, symbol);
+    // In case the identifier is part of a property access expression, we need to verify that the
+    // property access originates from a namespace that has been imported from Material or the CDK.
+    if (ts.isPropertyAccessExpression(identifier.parent)) {
+      const expression = identifier.parent.expression;
+
+      if (ts.isIdentifier(expression) && this.trustedNamespaces.has(expression.text)) {
+        return this._createFailureWithReplacement(identifier);
+      }
+    } else if (this.trustedIdentifiers.has(identifier.text)) {
+      return this._createFailureWithReplacement(identifier);
```

Comment:
**Note**: Surely this can be moved down to the end of the block and we just return if we do not want to proceed, but I think that way it's more clean.